src/main/java/com/cegeka/academy/service/userChallenge/UserChallengeServiceImpl.java

```diff
@@ -1,9 +1,129 @@
 package com.cegeka.academy.service.userChallenge;
 
+import com.cegeka.academy.domain.Challenge;
+import com.cegeka.academy.domain.Invitation;
+import com.cegeka.academy.domain.User;
+import com.cegeka.academy.domain.UserChallenge;
+import com.cegeka.academy.repository.UserChallengeRepository;
+import com.cegeka.academy.service.dto.*;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
 @Service
 @Transactional
 public class UserChallengeServiceImpl implements UserChallengeService {
+
+    @Autowired
+    private UserChallengeRepository userChallengeRepository;
+
+    @Override
+    public List<UserChallengeDTO> getUserChallengesByUserId(Long userId) {
+
+        List<UserChallenge> userChallengeList = userChallengeRepository.findAllByUserId(userId);
+
+        return userChallengeList.stream().map(userChallenge -> convertUserChallengeToUserChallengeDTO(userChallenge)).collect(Collectors.toList());
+
+    }
+
+    public static UserChallenge convertUserChallengeDTOToUserChallenge(UserChallengeDTO userChallengeDTO){
+
+        if(userChallengeDTO != null){
+
+            Invitation invitation = new Invitation();
+            invitation.setInvitedUser(convertUserForUserChallengeDTOToUser(userChallengeDTO.getInvitation().getUser()));
```

Comment:
fiecare conversie ar trebui sa fie intr-o metoda separata. Avand in vedere ca avem multe obiecte de convertit, ar trebui sa definesti o clasa separata in care sa tii aceste metode. De exemplu: UtilMapper